src/main/java/edu/wpi/grip/core/Socket.java

```diff
@@ -13,38 +12,39 @@
 import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
- * A socket is a wrapper for a value that can be updated and passed around operations.  Sockets contain a set of hints
+ * A Socket is an abstract wrapper for a value that can be updated and passed around operations.  Sockets contain a set of hints
  * about the data they contain, as well as an actual value.
  * <p>
- * Sockets that are given to operations are referred to as "input sockets", and sockets that operations store their
- * results in are referred to as "output sockets".
+ * Sockets that are given to operations are {@link InputSocket InputSockets}, and sockets that operations store their
+ * results are {@link OutputSocket OutputSockets}.
+ *
+ * @param <T> The type of the value that this socket stores
  */
-public class Socket<T> {
+public abstract class Socket<T> {
     public enum Direction {INPUT, OUTPUT}
 
     private final EventBus eventBus;
```

Comment:
This should be made protected, right?  For when subclasses need to send events. (This isn't leaking anything since almost every class in the application has access to the same eventbus)